### ISSUE TITLE
unix socket support

### ISSUE THREAD

**Issue by ghost** [amphp/uri#9]
I try:

```unix:///tmp/redis.sock```
```unix:/tmp/redis.sock```

Looks like it is not supported.

**Comment by bwoebi** [comment]
The latter one is not valid as per RFC 3986, but RFC 3986 allows for an empty hostname (reg-name), thus `unix:///tmp/redis.sock` must be supported here.

**Comment by bwoebi** [comment]
Further note is that we are using `parse_url()` internally, which doesn't work really well a bunch of things qualifying as URI.

From its documentation:
```
Note:
This function is intended specifically for the purpose of parsing URLs and not URIs. However, to comply with PHP's backwards compatibility requirements it makes an exception for the file:// scheme where triple slashes (file:///...) are allowed. For any other scheme this is invalid.
```

---

Also, we are here generally matching a bit of a mixture between URI and URL currently … we ought cleaning that up…

**Comment by ghost** [comment]
@bwoebi I think that here: https://github.com/amphp/socket/blob/master/lib/functions.php#L70 must be used ```amphp/uri```

**Comment by kelunik** [comment]
We won't add any new features to this library. Please use https://github.com/thephpleague/uri-parser instead.

**Comment by exochron** [comment]
Hi there. I stumbled upon this issue while searching/trying to connect amphp/redis via unix socket.
Currently (redis v0.3.3/ uri v0.1.4) I'm having the weird behaviour that the first folder is basically truncated. Therefore the client can't find the file. So I just added an extra folder in my uri. ^^
File: /tmp/redis/redis.sock
URI: unix://tmp/tmp/redis/redis.sock
I hope that helps another lost soul from the internet. :)
